site/src/pages/docs/server-timeouts.mdx

```diff
@@ -0,0 +1,210 @@
+# Overriding server timeouts
+
+You can override the default timeouts of a request.
+
+## Global Configuration
+
+Globally configuring the default timeout with system properties or a custom <type://FlagsProvider>.
+
+### Using JVM system properties
+
+You can override the default server timeouts by specifying the following JVM system properties if you do not prefer setting it programmatically.
+
+`-Dcom.linecorp.armeria.defaultRequestTimeoutMillis=<long>`
+
+The default value of this flag is 10000L and 0 disables the timeout.
+
+<Warning>
+
+  The JVM system properties have effect only when you did not specify them programmatically.
+  See <type://Flags> for the complete list of JVM system properties in Armeria.
+
+</Warning>
+
+### Using <type://FlagsProvider>
+
+You can override the global <type://Flags> by implementing your own <type://FlagsProvider> and loading it via Java SPI.
+
+```java
+package com.example.providers;
+
+// Add the following text file to your classpath or JAR file:
+//
+// $ cat META-INF/services/com.linecorp.armeria.common.FlagsProvider
+// com.example.providers.MyFlagsProvider
+//
+public class MyFlagsProvider implements FlagsProvider {
+  @Override
+  public int priority() {
+    // The provider with higher value will be evaluated first.
+    return 100;
+  }
+
+  @Override
+  public Long defaultRequestTimeoutMillis() {
+      return 1000L;
+  }
+}
+```
+
+## Using <type://ServerBuilder>
+```java
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+import java.time.Duration;
+
+int requestTimeout = 5;
+
+ServerBuilder sb = Server.builder();
+Server server = sb.http(port)
+                  .annotatedService(new HelloService())
+                  .requestTimeout(Duration.ofSeconds(requestTimeout))
+                  .build();
+```
+
+## Using <type://VirtualHostBuilder>
+
+You can set the timeouts for each virtual host individually
+
+```java
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+import java.time.Duration;
+
+import static com.linecorp.armeria.common.HttpStatus.OK;
+
+int requestTimeout1 = 5;
+int requestTimeout2 = 10;
+int requestTimeout3 = 20;
```

Comment:
Optional) Personally I think inlining the variables makes the code more concise/readable. (ditto for the other examples)